docs/YOSEMITE.md

```diff
@@ -0,0 +1,103 @@
+
+`StoresManager` is a singleton. 
+
+The `AuthenticateState`aggregates a reference to `Dispatcher`. This is an important piece of the system, as it is responsible for executing the actions declared in the different Stores. Therefore we will discuss it in depth later on.
+
+## Domain level concerns. Store
+As mentioned in the previous section, the business logic pertaining each domain level concern is encapsulated in a separate subclass of `Store` .
+
+`Store` is an implementation of the `ActionsProcessor` protocol. We will discuss this protocol later, but for now, let’s say it is _something that can process an action_.
+
+`Store` is injected with a reference to a `Dispatcher`, an implementation of the  `StorageManagerType`  protocol (the storage stack) and an implementation of the `Network` protocol (the networking stack).
+
+As a quick aside, please note this pattern: dependencies are usually injected.
+
+`Store` is a lightweight implementation of the Template Method Pattern. It enforces its subclasses to implement a method  (called `registerSupportedActions`) that will register a collection of _supported actions_ with the `Dispatcher` and to implement a method that deals with incoming actions.  We will discuss what actions are and how they are modelled a bit later.
+
+Each subclass of `Store` registers a processor with the `Dispatcher`. To register a processor, a subclass of `Store`overrides `registerSupportedActions`. In the implementation of that method, it will register itself as processor, for a given set of actions.
+
+That set of actions is declared in an implementation of the `Action` protocol. We will look at `Action` in details in the next section, but for now,  it is important to note that there is a one to one relationship between each subclass os `Store` and a corresponding implementation of the `Action` protocol. (i.e. for `AccountStore` we would also have a `AccountAction`).
+
+At the moment, we provide the following subclasses of `Store`:
+* `AccountStore`. Registers `AccountAction` with the `Dispatcher`.  It implements the business logic necessary  to manage an account (load an account, load a site, synchronise account information…)
+* `CommentStore`.  Registers and responds to actions declared in `CommentAction` It implements the business logic pertaining comments.
+* `NotificationStore`. Registers  and responds to`NotificationAction` 
+* `OrderStore`. Registers `OrderAction` with the `Dispatcher` and responds to the actions declared in it.
+* `OrderNoteStore`. Business logic pertaining order notes. Registers and responds to operation declared in `OrderNoteAction`
+* `OrderStatusStore`. Business logic related to order statuses, implementing operations declared in `OrderStatusAction`
```

Comment:
specifically, `order payment statuses`. I'm not sure if the distinction that the statuses are for **payment** helps or not, though. Your call 😀